packages/next-swc/crates/next-core/js/src/entry/app-renderer.tsx

```diff
@@ -290,6 +290,7 @@ async function runOperation(renderData: RenderData) {
   return {
     headers: [
       ["Content-Type", result.contentType() ?? MIME_TEXT_HTML_UTF8],
+      ["Vary", "RSC"],
```

Comment:
All provided vary values need to be included as otherwise the browser cache will be messed up when prefetching / when the input router state is different.

Reply:
Done.